engine/hid/src/hid_null.cpp

```diff
@@ -25,8 +25,13 @@ namespace dmHID
     // detect sloppy init/final usage
     dmHashTable<uintptr_t, char*>* g_DummyData = 0x0;
 
+    // Mostly used for tests
+    static uint8_t g_MouseVisible = 0;
+
```

Comment:
I think it's an interesting problem. (I don't propose a change to this behavior)

We currently don't have a good way of having test code inside a library.
And at the same time, to what extent do we want our null implementations to behave like a non-null version?
I think it's important to remember that both graphics/hid null are used on headless servers.

E.g. does it make sense for the hid_null to report 0 devices, but still report mouse visibility?

Perhaps if we in our tests could register some null devices (keyboard, gamepad etc) to test functionality against.

Reply:
Yes it’s tricky. In this case I want to make sure api is unit tested, but the functionality doesn’t really do anything. Not sure how a “null” mouse would work 🧐

Speaking of null/headless, In reality I would like to be able to run tests on the actual graphics backend which is guess we can’t do on our GH CI right now.

Reply:
In my mind, a custom device would produce input from whatever the source would be.
It could be controlled from the unit test, to produce clicks and movements, it could also be controlled via a network connection, letting the user type commands into the application etc.